.eslintrc.json

```diff
@@ -0,0 +1,136 @@
+{
+  "root": true,
+
+  "env": {
+    "node": true
+  },
+
+  "globals": {},
+
+  "rules": {
+    // Rules are divided into sections from http://eslint.org/docs/rules/
+
+    // Possible errors
+    "comma-dangle": ["error", "never"],
+    "no-cond-assign": ["error", "except-parens"],
+    "no-constant-condition": "error",
+    "no-control-regex": "error",
+    "no-debugger": "error",
+    "no-dupe-args": "error",
+    "no-dupe-keys": "error",
+    "no-duplicate-case": "error",
+    "no-empty-character-class": "error",
+    "no-empty": "error",
+    "no-ex-assign": "error",
+    "no-extra-boolean-cast": "error",
+    "no-extra-semi": "error",
+    "no-func-assign": "error",
+    "no-inner-declarations": "error",
+    "no-invalid-regexp": "error",
+    "no-irregular-whitespace": "error",
+    "no-negated-in-lhs": "error",
+    "no-obj-calls": "error",
+    "no-regex-spaces": "error",
+    "no-sparse-arrays": "error",
+    "no-unreachable": "error",
+    "use-isnan": "error",
+    "no-unsafe-finally": "error",
+    "valid-typeof": "error",
+    "no-unexpected-multiline": "error",
+
+    // Best practices
+    "accessor-pairs": "error",
+    "array-callback-return": "error",
+    "complexity": ["warn", 10],
+//    "dot-notation": "error", // TODO enable?
```

Comment:
I was thinking about moving all those `TODO` rules to a separate
.eslintrc-todo.json file, then discuss all the options, remove the one we
don't want to enable and the rest will stay there, waiting for someone to
fix the code and enable the rule.

Thoughts?

On Tue, Jul 26, 2016 at 1:32 PM, Georgios Kalpakas <notifications@github.com
> wrote:

> In .eslintrc.json
> <https://github.com/angular/angular.js/pull/14952#discussion_r72231300>:
>
> > +    "no-irregular-whitespace": "error",
> > +    "no-negated-in-lhs": "error",
> > +    "no-obj-calls": "error",
> > +    "no-regex-spaces": "error",
> > +    "no-sparse-arrays": "error",
> > +    "no-unreachable": "error",
> > +    "use-isnan": "error",
> > +    "no-unsafe-finally": "error",
> > +    "valid-typeof": "error",
> > +    "no-unexpected-multiline": "error",
> > +
> > +    // Best practices
> > +    "accessor-pairs": "error",
> > +    "array-callback-return": "error",
> > +    "complexity": ["warn", 10],
> > +//    "dot-notation": "error", // TODO enable?
>
> I think there are places where we deliberately use bracket-notation. If
> there are many such places (that need a "manual" override), I would leave
> this disabled.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular.js/pull/14952/files/6f56b890138f16a67b5f61cb2238d75378a76497..22f65cbbbf36f1ae396038f457ca2266f5b52767#r72231300>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABrUnvg_rt1IyxEc7jYUKkY9aFjn3a20ks5qZfBkgaJpZM4JU90D>
> .
>



-- 
Michał Gołębiowski